upgrade/upgrade-instructions-for-improved-data-fixtures-and-tests.md

```diff
@@ -0,0 +1,63 @@
+          }
+        ``` 
+- Modify your functional tests:
+    - Add `skipTestIfFirstDomainIsNotInEnglish`, `getReferenceForDomain`, and `getFirstDomainLocale` methods to your [`FunctionalTestCase`](https://github.com/shopsys/shopsys/blob/v8.1.0/project-base/tests/ShopBundle/Test/FunctionalTestCase.php).
+    - When accessing a translated attribute on first domain, use `FunctionalTestCase::getFirstDomainLocale` instead of hard-coding the locale.
+    - In tests that make sense only for the English locale (e.g. tests for searching a particular phrase), you can skip their execution for different settings using `FunctionalTestCase::skipTestIfFirstDomainIsNotInEnglish`.
+    - When asserting prices, use [`PriceConverter`](https://github.com/shopsys/shopsys/blob/v8.1.0/packages/framework/src/Model/Pricing/PriceConverter.php) class to convert the prices to proper currency.
+    - Use `ParameterFacade::getParameterValueByValueTextAndLocale()` to get the proper entity in tests where the `ParameterValue` ids are used.
+    - Modify your `app/config/packages/test/prezent_doctrine_translatable.yml` to use `%locale%` value for setting `fallback_locale` instead of hard-coding it.
+- Modify your acceptance tests:
+    - Add two new modules - [`LocalizationHelper`](https://github.com/shopsys/shopsys/blob/v8.1.0/project-base/tests/ShopBundle/Test/Codeception/Helper/LocalizationHelper.php) and [`NumberFormatHelper`](https://github.com/shopsys/shopsys/blob/v8.1.0/project-base/tests/ShopBundle/Test/Codeception/Helper/NumberFormatHelper.php)
+        - The modules add new methods to acceptance tester that are useful in the tests, e.g.:
+            - `amOnLocalizedRoute()`
+            - `seeTranslationAdmin()`
+            - `seeTranslationFrontend()`
+            - `getFormattedPriceWithCurrencySymbolOnFrontend()`
+        - Rewrite your tests to make them resistant against domains and locales settings using these methods.
+        - Register the new modules in your `acceptance.suite.yml` configuration in `modules.enabled` section.
+    - Update your [`Cart/index.html.twig`](https://github.com/shopsys/shopsys/blob/v8.1.0/project-base/src/Shopsys/ShopBundle/Resources/views/Front/Content/Cart/index.html.twig) - add JS class `js-cart-item-price` it is used for identification of item price in cart in acceptance tests.
+- Do not forget to update your [`services.yml`](https://github.com/shopsys/shopsys/blob/v8.1.0/project-base/src/Shopsys/ShopBundle/Resources/config/services.yml) and [`services_test.yml`](https://github.com/shopsys/shopsys/blob/v8.1.0/project-base/src/Shopsys/ShopBundle/Resources/config/services_test.yml).
+- Run `php phing translations-dump` and fill in new translation for your locale in `dataFixtures.xx.po` (`xx` stands for your locale).
+- Update your [`easy-coding-standard.yml`](https://github.com/shopsys/shopsys/blob/v8.1.0/project-base/easy-coding-standard.yml) (some of the data fixture classes need to be excluded from some checks).
```

Comment:
how should I update it?

Reply:
Well all upgrade notes at beginning say:
If any of the following instructions is not clear enough for you or you need more information, we recommend to take a look at the actual changes in #1425.